ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -114,7 +114,34 @@ public void start() throws StoreException {
 
   @Override
   public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> storeGetOptions) throws StoreException {
-    throw new UnsupportedOperationException("Method not supported");
+    checkStarted();
+    if (ids.size() > 1 && ids.size() != new HashSet<>(ids).size()) {
+      throw new IllegalArgumentException("The list of IDs provided contains duplicates");
+    }
+    List<BlobReadInfo> blobReadInfos = new ArrayList<>(ids.size());
+    List<MessageInfo> messageInfos = new ArrayList<>(ids.size());
+    try {
+      List<BlobId> blobIdList = ids.stream().map(key -> (BlobId) key).collect(Collectors.toList());
+      Map<String, CloudBlobMetadata> cloudBlobMetadataListMap = cloudDestination.getBlobMetadata(blobIdList);
+      if (cloudBlobMetadataListMap.size() < ids.size()) {
```

Comment:
I think that BlobStore's get method does return an exception if one of the keys is not found. Basically BlobStore::get calls PersistentIndex::getBlobReadInfo for each id, which in turn tries to find the id in the index. If the id is not found in the index, it seems to be throwing StoreException.

In our case it would be more efficient to get all the metadata for the list of ids in one cosmos call. If we do it in the CloudBlobStore::get() method, then we will know if the blobs are present, and be able to throw exception in case they aren't. This will be in line with the behavior of BlobStore::get. 

The actual CloudDestination::downloadBlob happens either during prefetch, or when the response if being returned to frontend.